src/nvim/buffer.c

```diff
@@ -1718,8 +1718,8 @@ static buf_T *buflist_findname_file_id(char_u *ffname, FileID *file_id,
  */
 int 
 buflist_findpat (
-    char_u *pattern,
-    char_u *pattern_end,       /* pointer to first char after pattern */
+    const char_u *pattern,
+    const char_u *pattern_end,       /* pointer to first char after pattern */
```

Comment:
The `const` changes here cascaded from `write_reg_contents_ex`. This is fine because `buflist_findpat` and `file_pat_to_reg_pat` do not modify the contents of their pointer params.